### ISSUE TITLE
"Delayed model change job" / "Index out of bounds" error when debugging Spring Boot app

### ISSUE THREAD

**Issue by hafarey** [spring-projects/sts4#453]
I'm encountering a "Delayed model change job" error when I debug a Spring Boot app within STS.  Attached is a screenshot of the error dialog, and below I've pasted the stacktrace from the STS Error Log Event Details dialog.

![image](https://user-images.githubusercontent.com/17915941/80654109-ed6d3d00-8abe-11ea-8bc5-bed4a62b6aa8.png)

This appears to be a bug carried over from Eclipse:  [Bug 561338](https://bugs.eclipse.org/bugs/show_bug.cgi?id=561338).  I'm reporting it here because it's affecting STS and my development workflow.

**My STS info**
```
macOS 10.13.6

STS Version: 4.6.1.RELEASE
Build Id: 202004200639

eclipse.buildId=4.6.1.202004200639-RELEASE
java.version=1.8.0_221
java.vendor=Oracle Corporation
```

**Stack trace**
```
java.lang.IllegalArgumentException: Index out of bounds
	at org.eclipse.swt.SWT.error(SWT.java:4704)
	at org.eclipse.swt.SWT.error(SWT.java:4638)
	at org.eclipse.swt.SWT.error(SWT.java:4609)
	at org.eclipse.swt.widgets.Widget.error(Widget.java:810)
	at org.eclipse.swt.widgets.TreeItem.getItem(TreeItem.java:740)
	at org.eclipse.jface.viewers.TreeViewer.lambda$4(TreeViewer.java:829)
	at org.eclipse.jface.viewers.StructuredViewer.preservingSelection(StructuredViewer.java:1397)
	at org.eclipse.jface.viewers.TreeViewer.preservingSelection(TreeViewer.java:363)
	at org.eclipse.jface.viewers.StructuredViewer.preservingSelection(StructuredViewer.java:1358)
	at org.eclipse.jface.viewers.TreeViewer.remove(TreeViewer.java:810)
	at org.eclipse.debug.internal.ui.viewers.model.TreeModelContentProvider.handleRemove(TreeModelContentProvider.java:1546)
	at org.eclipse.debug.internal.ui.viewers.model.TreeModelContentProvider.updateNodes(TreeModelContentProvider.java:1266)
	at org.eclipse.debug.internal.ui.viewers.model.TreeModelContentProvider.updateNodes(TreeModelContentProvider.java:1299)
	at org.eclipse.debug.internal.ui.viewers.model.TreeModelContentProvider.updateNodes(TreeModelContentProvider.java:1299)
	at org.eclipse.debug.internal.ui.viewers.model.TreeModelContentProvider.updateNodes(TreeModelContentProvider.java:1299)
	at org.eclipse.debug.internal.ui.viewers.model.TreeModelContentProvider.updateModel(TreeModelContentProvider.java:562)
	at org.eclipse.debug.internal.ui.viewers.model.TreeModelContentProvider.doModelChanged(TreeModelContentProvider.java:530)
	at org.eclipse.debug.internal.ui.viewers.model.TreeModelContentProvider.access$0(TreeModelContentProvider.java:524)
	at org.eclipse.debug.internal.ui.viewers.model.TreeModelContentProvider$DelayedDoModelChangedJob.runInUIThread(TreeModelContentProvider.java:436)
	at org.eclipse.ui.progress.UIJob.lambda$0(UIJob.java:95)
	at org.eclipse.swt.widgets.RunnableLock.run(RunnableLock.java:40)
	at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:185)
	at org.eclipse.swt.widgets.Display.runAsyncMessages(Display.java:4126)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3793)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1160)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1049)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:155)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:658)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:557)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:154)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:150)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:203)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:137)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:107)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:401)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:255)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:657)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:594)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1447)
```

**Comment by martinlippert** [comment]
From looking at the Eclipse bugs and reading through the comments, it looks like this is already fixed in Eclipse 4.16M1. In addition to that there is a workaround mentioned how you can avoid the issue. Does that workaround work for you?

You could also go ahead and grab a latest CI build of Spring Tools 4 based on the latest milestone builds of Eclipse 4.16 (2020-06), which should already contain the fix - if you wanna give that a try.

**Comment by martinlippert** [comment]
The latest releases of Spring Tools 4 for Eclipse should already contain the fix since the Eclipse 2020-06 release (from June 2020), so closing this as a duplicate of the Eclipse one. If you come across something like this again, please comment and we can re-open this, of course.
